chapter/background.md

```diff
@@ -0,0 +1,607 @@
+
+Nickel [@nickel], the language targeted by the language server detailed in this thesis, defines itself as "configuration language" used to automize the generation of static configuration files.
+
+Static configuration languages such as XML[@xml], JSON[@json], or YAML[@yaml] are language specifications defining how to textually represent structural data used to configure parameters of a system^[some of the named languages may have been designed as a data interchange format which is absolutely compatible with also acting as a configuration language].
+Applications of configuration languages are ubiquitous especially in the vicinity of software development. While XML and JSON are often used by package managers [@npm, @maven, @composer], YAML is a popular choice for complex configurations such as CI/CD pipelines [@travis, @ghaction, @gitlab-runner] or machine configurations in software defined networks such as Kubernetes and docker compose.
+
+Such static formats are used due to some significant advantages compared to other formats.
+Most strikingly, the textual representation allows inspection of a configuration without the need of a separate tool but a text editor and be version controlled using VCS software like Git.
+For software configuration this is well understood as being preferable over databases or other binary formats. Linux service configurations (files in `/etc`) and MacOS `*.plist` files which can be serialized as XML or a JSON-like format, especially exemplify that claim.
+
+Yet, despite these formats being simple to parse and widely supported [@json], their static nature rules out any dynamic content such as generated fields, functions and the possibility to factorize and reuse.
+Moreover, content validation has to be developed separately, which led to the design of complementary schema specification languages like json-schema [@json-schema] or XSD [@xsd].
+
+These qualities require an evaluated language.
+In fact, some applications make heavy use of config files written in the native programming language which gives them access to language features and existing analysis tools.
+Examples include JavaScript frameworks such as webpack [@webpack] or Vue [@vue] and python package management using `setuptools`[@setuptools].
+
+Despite this, not all languages serve as a configuration language, e.g. compiled languages and some domains require language agnostic formats.
+For particularly complex products, both language independence and advanced features are desirable.
+Alternatively to generating configurations using high level languages, this demand is addressed by more domain specific languages.
+Dhall [@dhall], Cue [@cue] or jsonnet [@jsonnet] are such domain specific languages (DSL), that offer varying support for string interpolation, (strict) typing, functions and validation.
+
+### Infrastructure as Code
+
+A prime example for the application of configuration languages are IaaS^[Infrastructure as a Service] products.
```

Comment:
Do all configuration languages feat under IaC? Is IaC a superset or a subset of configuration languages?

Make a transition for the reader.